operator/api/v1alpha1/service_types.go

```diff
@@ -17,22 +17,30 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"strings"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 )
 
+type Owner struct {
+	UID  types.UID `json:"uid"`
+	Kind string    `json:"kind"`
+}
+
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // ServiceSpec defines the desired state of Service
 type ServiceSpec struct {
-	ApplicationName   string    `json:"application,omitempty"`
-	PreDeplymentCheck EventSpec `json:"preDeploymentCheck"`
+	ApplicationName    string    `json:"application"`
+	Version            string    `json:"version"`
+	PreDeploymentCheck EventSpec `json:"preDeploymentCheck"`
+	Owner              Owner     `json:"owner"`
```

Comment:
I would rename this to something else than `Owner`, as the referenced K8s object is not a owner in the usual K8s sense (owner being the one creating the instance of that CRD). Maybe something like `ReplicasetReference` would be more suited... @thschue what do you think?

Reply:
in any case, please add a comment regarding to what this property refers to

Reply:
I think ReplicaSetReference is fine, owner is misleading in this case ...

Reply:
Can we go with `ResourceReference`?